Source/Core/Core/PowerPC/JitArm64/Jit.cpp

```diff
@@ -363,18 +363,35 @@ void JitArm64::WriteExit(u32 destination, bool LK, u32 exit_address_after_return
 
   if (LK)
   {
-    // Push {ARM_PC+20; PPC_PC} on the stack
+    // Push {ARM_PC; PPC_PC} on the stack
     MOVI2R(ARM64Reg::X1, exit_address_after_return);
-    ADR(ARM64Reg::X0, 20);
+    ADR(ARM64Reg::X0, JitArm64BlockCache::BLOCK_LINK_FAST_BL_OFFSET + 12);
     STP(IndexType::Pre, ARM64Reg::X0, ARM64Reg::X1, ARM64Reg::SP, -16);
   }
 
+  constexpr size_t farcode_size = 3 * sizeof(u32);
+  const bool switch_to_far_code = !IsInFarCode();
+  const u8* farcode_1_addr;
+  if (switch_to_far_code)
+  {
+    SwitchToFarCode();
```

Comment:
Wouldn't farcode_1_addr = m_far_code.GetCodePtr() be simpler? Also not sure if there are more descriptive names than farcode_1_addr and farcode_2_addr?

Reply:
I didn't think of `m_far_code.GetCodePtr()`. But unfortunately, after testing it, it doesn't actually work for what we want to do. Calling `SwitchToFarCode` will align the current farcode pointer to 16 bytes, and we need to take that into account when obtaining the address.

I've renamed the variables.